[Kim Yongbeom] iP

Yongbeom-Kim opened the pull request:
# DukePro
“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))
DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ _SUPER_ FAST to use

## HOW TO

All you need to do is,
download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
1. double-click it.
2. add your tasks.
3. let it manage your tasks for you 😉
4. And it is **FREE**!

Features:
- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

---

If you Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Yongbeom-Kim committed:
changed cli banner

Yongbeom-Kim committed:
echo command + bye command

Yongbeom-Kim committed:
echo command + bye command

Yongbeom-Kim committed:
rename conversation handler file

Yongbeom-Kim committed:
Store all texts in a list, and enuemrate them with list

Yongbeom-Kim committed:
added tasks class + done/not done functionality

Yongbeom-Kim committed:
mark command, set done

Yongbeom-Kim committed:
added ability to unmark tasks

Yongbeom-Kim committed:
Polymorphism (events, deadlines, todos) from tasks

Yongbeom-Kim committed:
basic tests with stdio redirection

Yongbeom-Kim committed:
added delete/remove functionality

Yongbeom-Kim committed:
added a command parser for all commands

Yongbeom-Kim committed:
done error catching

Yongbeom-Kim committed:
remove .class files

Yongbeom-Kim committed:
basic fileio util class

teekaytai commented on src/main/java/ConversationHandler.java:
```diff
@@ -0,0 +1,181 @@
+
+                default:
+                    return input;
+            }
```
Perhaps the case clauses can be aligned with the switch statement instead of indented?

teekaytai commented on src/main/java/ConversationHandler.java:
```diff
@@ -0,0 +1,181 @@
+    }
+
+    public String commandHandler(String input) {
+//        We get the first word, since that determines the command
```
Comments could perhaps be indented relative to the surrounding code to help with readability

teekaytai commented on src/main/java/ConversationHandler.java:
```diff
@@ -0,0 +1,181 @@
+public class ConversationHandler {
+
+    private Scanner in;
+    private boolean active = true;
```
Perhaps changing the name `active` to `isActive` would make it clearer to readers that this variable is a boolean

teekaytai commented on src/main/java/ConversationHandler.java:
```diff
@@ -0,0 +1,181 @@
+        this.list.remove(index);
+        return "Noted. I've removed this task: \n" +
+                task.toString() + "\n" +
+                "Now you have " + this.list.size() + " tasks in the list.";
```
I like that you gave these lines an indentation of 8 spaces. It helps emphasise that they are wrapped lines

teekaytai commented on src/main/java/utils/InputParser.java:
```diff
@@ -0,0 +1,56 @@
+        return intPattern.matcher(s).matches();
+    }
+
+    private static Pattern flagPattern = Pattern.compile("/\\w+");
+    /**
```
Maybe a blank line could be added between the variable and the javadoc for the method below to separate logical units

teekaytai commented on src/main/java/task_classes/Task.java:
```diff
@@ -0,0 +1,28 @@
+        return "[" + this.getStatusIcon() + "] " + this.description;
+    }
+
+    public void setDone(){
```
I think it would be nice to have a space between the parentheses for the arguments and the curly brace

teekaytai reviewed (commented):
Overall the code is neat and readable. Just some minor improvements could be done to help make the code even cleaner!

nickeltea commented on src/main/java/task_classes/Task.java:
```diff
@@ -0,0 +1,28 @@
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
```
Good job keeping methods short and readable!

nickeltea commented on src/main/java/ConversationHandler.java:
```diff
@@ -0,0 +1,181 @@
+                "Now you have " + list.size() + " tasks in the list.";
+    }
+
+    private String addDeadlineCommand(HashMap<String, String> commandObj) {
```
Good job keeping the execution path clear and prominent!

nickeltea commented on src/main/java/ConversationHandler.java:
```diff
@@ -0,0 +1,181 @@
+
+    }
+
+    private String addTodoCommand(HashMap<String, String> commandObj) {
```
I like that you used an explanatory name for this method!

nickeltea commented on src/main/java/ConversationHandler.java:
```diff
@@ -0,0 +1,181 @@
+    public String commandHandler(String input) {
+//        We get the first word, since that determines the command
+        HashMap<String, String> command = InputParser.getInputArguments(input);
+        try {
```
I like how you avoided deep nesting here.

nickeltea commented on src/main/java/ConversationHandler.java:
```diff
@@ -0,0 +1,181 @@
+    private boolean active = true;
+    private ArrayList<Task> list = new ArrayList<>();
+
+    public ConversationHandler() {
```
I like that your code is very readable.

Yongbeom-Kim committed:
misc changes

Yongbeom-Kim committed:
gradle build

Yongbeom-Kim committed:
refactored conversation handler to move scanner to main

Yongbeom-Kim committed:
added json conversion methods to task classes

Yongbeom-Kim committed:
resolve gitignore conflict

Yongbeom-Kim committed:
added file write method:

Yongbeom-Kim committed:
file writes automatically on task list edit

Yongbeom-Kim committed:
Added save and load functionality

Yongbeom-Kim committed:
added dates

Yongbeom-Kim committed:
Merge branch 'branch-level-8' + debug

Yongbeom-Kim committed:
Moved all print statements to UI class

Yongbeom-Kim committed:
Implemented command class

Yongbeom-Kim committed:
Refactored Main class into Command and Parser

Yongbeom-Kim committed:
Create tasklist class

Yongbeom-Kim committed:
Move scanner to UI class

Yongbeom-Kim committed:
Fix scanner nullpointer

Yongbeom-Kim committed:
Add close scanner functionality to UI

Yongbeom-Kim committed:
Refactored local storage with LocalStorage class

Yongbeom-Kim committed:
delete Main

Yongbeom-Kim committed:
Package, split classes into relevant packages

Yongbeom-Kim committed:
Changed Ui variables privacy

Yongbeom-Kim committed:
Added test: command argument parser

Yongbeom-Kim committed:
added JavaDoc to all classes

Yongbeom-Kim committed:
code base follows coding standard

Yongbeom-Kim committed:
Add find command to search tasks

Yongbeom-Kim committed:
Merge branch 'branch-A-JavaDoc', fix merge conflicts

Yongbeom-Kim committed:
Merge branch 'branch-A-CodingStandard', fixed merge conflicts

Yongbeom-Kim committed:
Added javafx fxml files

Yongbeom-Kim committed:
Integrate javaFX UI with backend

Yongbeom-Kim committed:
Cleaned dead code

Yongbeom-Kim committed:
Add assert statements

Yongbeom-Kim committed:
Standardize code quality and remove dead code

There are several methods and class attributes that are not used anymore,
and hence they have been gotten rid of.

Getting rid of these improves code quality.

As a step towards maintainablity, let's remove these to keep code cleaner.

Yongbeom-Kim committed:
Empty commit
I am very very sorry, I forgot to do this on a new branch

Yongbeom-Kim committed:
Again, very sorry
Forgot to do this on a new branch

Yongbeom-Kim committed:
Merge pull request #2 from Yongbeom-Kim/branch-A-CodeQuality

Again, very sorry

Yongbeom-Kim committed:
rudimentary first pane

Yongbeom-Kim committed:
Complete most UX elements

Yongbeom-Kim committed:
Complete rudimentari UI elements & stying

Yongbeom-Kim committed:
Add ability to update task list dynamically

Yongbeom-Kim committed:
Add scrollbar(pane) in task list pane

Yongbeom-Kim committed:
Show help screen + esc to close

Yongbeom-Kim committed:
Add notification pane for command feedback

Yongbeom-Kim committed:
Staged helpwindow fxml file

Yongbeom-Kim committed:
Add find to help button

Yongbeom-Kim committed:
Find book UI functionality

Yongbeom-Kim committed:
Indexing and better command functionality

Yongbeom-Kim committed:
Better null handling, especially when list has 0 items.

Yongbeom-Kim committed:
Format command text field + autofocus on text field

Yongbeom-Kim committed:
Add background to selected label

Yongbeom-Kim committed:
Fix nullpointer from deselecting task label

Yongbeom-Kim committed:
Organise CSS file

Yongbeom-Kim committed:
Rename CSS file to styles.css

Yongbeom-Kim committed:
Delete redundant files

Yongbeom-Kim committed:
Style scrollbars

Yongbeom-Kim committed:
Style scrollbars credit

Yongbeom-Kim committed:
Style scrollbars credit

Yongbeom-Kim committed:
Style scrollbars credit

Yongbeom-Kim committed:
Add assertions

Yongbeom-Kim committed:
Merge branch 'branch-A-Assertions'

Yongbeom-Kim committed:
Sorry, I forgot about the pull request thing

Yongbeom-Kim committed:
Merge pull request #3 from Yongbeom-Kim/branch-A-Assertions

Sorry, I forgot about the pull request thing

Yongbeom-Kim committed:
Update help window

Yongbeom-Kim committed:
Added User Guide

Yongbeom-Kim committed:
Added ui screenshot

Yongbeom-Kim committed:
Fix table rendering

Yongbeom-Kim committed:
oops wrong pic

Yongbeom-Kim committed:
Add product screenshot to user guide

Yongbeom-Kim committed:
Update README.md

Yongbeom-Kim committed:
Fix some bugs

Yongbeom-Kim committed:
Update UI Screenshot

Yongbeom-Kim committed:
Merge branch 'master' of https://github.com/Yongbeom-Kim/ip

Yongbeom-Kim committed:
Remove print

Yongbeom-Kim committed:
Update javafx to 11.0.2 (bugfix for ubuntu)

Yongbeom-Kim committed:
Make GUI exit on by command
